src/main/java/org/alfresco/util/shard/ExplicitShardingPolicy.java

```diff
@@ -67,6 +69,12 @@ public boolean configurationIsValid()
             {
                 if (test % numNodes == nodeInstance - 1)
                 {
+                    // This algorithm fails for some sets of parameters. (See SEARCH-1785)
+                    if (shardIds.contains(shard % numShards))
+                    {
+                        throw new AlfrescoRuntimeException("Error while attempting to create shard list for node " + nodeInstance
+                                + " (shards:" + numShards + ", replication:" + replicationFactor + ", nodes:" + numNodes + ")");
+                    }
```

Comment:
We need to add a more explicit error message, to inform the user that this configuration is not valid.